tracer/src/Datadog.Trace/PlatformHelpers/AspNetCoreHttpRequestHandler.cs

```diff
@@ -108,22 +108,16 @@ public Scope StartAspNetCorePipelineScope(Tracer tracer, Security security, Http
 
             SpanContext propagatedContext = ExtractPropagatedContext(request);
 
-            AspNetCoreTags tags;
-            if (tracer.Settings.RouteTemplateResourceNamesEnabled)
-            {
-                var originalPath = request.PathBase.HasValue ? request.PathBase.Add(request.Path) : request.Path;
-                httpContext.Features.Set(new RequestTrackingFeature(originalPath));
-                tags = new AspNetCoreEndpointTags();
-            }
-            else
-            {
-                tags = new AspNetCoreTags();
-            }
+            var routeTemplateResourceNames = tracer.Settings.RouteTemplateResourceNamesEnabled;
+            var tags = routeTemplateResourceNames ? new AspNetCoreEndpointTags() : new AspNetCoreTags();
 
             var scope = tracer.StartActiveInternal(_requestInOperationName, propagatedContext, tags: tags);
             scope.Span.DecorateWebServerSpan(resourceName, httpMethod, host, url, userAgent, tags);
             AddHeaderTagsToSpan(scope.Span, request, tracer);
 
+            var originalPath = request.PathBase.HasValue ? request.PathBase.Add(request.Path) : request.Path;
+            httpContext.Features.Set(new RequestTrackingFeature(originalPath, scope.Span));
```

Comment:
Previously we were _only_ setting the feature if we were using the new resource names (the default), but now we _always_ need the tracking feature so we can access the root span

Reply:
So I'm not really sure, but could it be that this line is still using `var span = tracer.InternalActiveScope?.Span;` instead of the tracking?

https://github.com/DataDog/dd-trace-dotnet/blob/6d05977252daaeaafda933c952cbb1d98745c0d6/tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs#L686

Reply:
Good catch 😃 AFAICT there's nothing in that code path that explicitly _requires_ it's the root span, except that we potentially set the http code, so I think we _should_ use it. Thanks!

Reply:
Sounds good!
I somehow put this comment on the wrong comment as well - it was supposed to be on this one https://github.com/DataDog/dd-trace-dotnet/pull/4920#discussion_r1403428975 regarding the exception getting caught by `custom_post_routing`, but that seems to not be the cause of it